src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,150 @@
+                    currentNode.next = new Node<>(keyHashCode, key, value, null);
+                    size++;
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        }
 
+        if ((float) size / table.length > DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+        int keyHashCode = key.hashCode();
+        int index = Math.abs(keyHashCode % table.length);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            return null;
+        }
+        if (Objects.equals(currentNode.key,key)) {
+            return currentNode.value;
+        }
+        while (currentNode.next != null) {
+            currentNode = currentNode.next;
+            if (Objects.equals(currentNode.key,key)) {
+                return currentNode.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * 2];
+        size = 0;
+
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private void putForNullKey(V value) {
+        if (table[0] == null) {
+            table[0] = new Node<>(0, null, value, null);
+            size++;
+        } else {
+            Node<K, V> currentNode = table[0];
+            while (true) {
+                if (currentNode.key == null) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(0, null, value, null);
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+    }
+
+    private V getForNullKey() {
+        Node<K, V> currentNode = table[0];
+        while (currentNode != null) {
+            if (currentNode.key == null) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public final String toString() {
+            return key + "=" + value;
+        }
+
+        public final int hashCode() {
+            return hash;
```

Comment:
The `hashCode` method in the `Node` class is redundant because it simply returns the precomputed hash value. Consider removing this method unless it's necessary for a specific reason.